src/main/java/net/minecraftforge/client/renderer/transparency/OITLevelRenderer.java

```diff
@@ -0,0 +1,515 @@
+    }
+
+    /**
+     * Indicates if the OIT rendering for level rendering is being handled.
+     * <p>
+     * When true, then render calls can be captured for transparent render types which require sorting.
+     *
+     * @return True, when level rendering is active, false when not.
+     */
+    public boolean isLevelRenderingInProgress()
+    {
+        return levelRenderingInProgress;
+    }
+
+    /**
+     * Sets the OIT rendering indicator for level rendering.
+     *
+     * @param levelRenderingInProgress True enables OIT rendering capture for level rendering, false disables it.
+     */
+    private void setLevelRenderingInProgress(boolean levelRenderingInProgress)
+    {
+        this.levelRenderingInProgress = levelRenderingInProgress;
+    }
+
+    /**
+     * Performs a check if the given cpu bound buffer render type is used during level rendering and requires transparency sorting.
+     * <p>
+     * If this is the case then the render call is postponed and queued up to be rendered through OIT.
+     * If not then the render is immediately queued on the GPU.
+     *
+     * @param renderType The render type with which the rendering should happen.
+     * @param renderedBuffer The CPU side payload buffer.
+     */
+    public void checkHandlesThenQueueOrRender(RenderType renderType, BufferBuilder.RenderedBuffer renderedBuffer)
+    {
+        final IRenderTypeBasedRenderCall call = new RenderedBufferRenderCall(renderType, renderedBuffer, new Matrix4f(RenderSystem.getModelViewMatrix()), new Matrix4f(RenderSystem.getProjectionMatrix()));
+        checkHandlesThenQueueOrRender(call);
+    }
+
+    /**
+     * Performs a check if the given chunk geometry render type is used during level rendering and requires transparency sorting.
+     * <p>
+     * If this is the case then the render call is postponed and queued up to be rendered through OIT.
+     * If not then the render is immediately queued on the GPU.
+     *
+     * @param renderType The render type with which the rendering should happen.
+     * @param currentPoseStack The current pose stack.
+     * @param cameraX The camera X position.
+     * @param cameraY The camera Y position.
+     * @param cameraZ The camera Z position.
+     * @param projectionMatrix The projection matrix.
+     */
+    public void checkHandlesThenQueueOrRender(RenderType renderType, PoseStack currentPoseStack, double cameraX, double cameraY, double cameraZ, Matrix4f projectionMatrix)
+    {
+        //We make a defensive copy so that the caller can't modify the matrix after we've captured it.
+        final IRenderTypeBasedRenderCall call = new ChunkGeometryRenderCall(renderType, Util.make(() -> {
+            final PoseStack stack = new PoseStack();
+            stack.last().pose().set(currentPoseStack.last().pose());
+            stack.last().normal().set(currentPoseStack.last().normal());
+            return stack;
+        }), cameraX, cameraY, cameraZ, Util.make(() -> {
```

Comment:
Given this is used in rendering so likely is hit quite a few times it may be worth just creating the PoseStack that is passed to the ChunkGeometryRenderCall in the body of this method rather than doing so using `Util#make` and a capturing lambda.

Reply:
Good idea